server/src/main/java/org/apache/druid/client/ImmutableDruidDataSource.java

```diff
@@ -120,8 +120,32 @@ public String toString()
            + "'}";
   }
 
+  /**
+   *
+   * Look at https://github.com/apache/incubator-druid/issues/7858 for explanation of why equals method is not supported.
```

Comment:
Please describe the reason properly why this method shouldn't be used in addition to the link to the original issue.

Reply:
Same for `hashCode`

Reply:
Done